proto/prysm/v1alpha1/attestation/id.go

```diff
@@ -22,12 +21,12 @@ const (
 )
 
 // Id represents an attestation ID. Its uniqueness depends on the IdSource provided when constructing the Id.
-type Id [33]byte
+type Id [32]byte
 
 // NewId --
 func NewId(att ethpb.Att, source IdSource) (Id, error) {
-	if err := helpers.ValidateNilAttestation(att); err != nil {
-		return Id{}, err
+	if att.IsNil() {
```

Comment:
Just to document somewhere that I dislike this function, it uses a string method separating with commas for no reason that I can see, on a hot path.

Reply:
I tried using bytes instead of strings and came up with this code (because the size of `int` is machine-dependent, I have to convert it to a fixed-size integer - `binary.Write` works only with fixed-size data):
```
indicesUint := make([]uint16, len(committeeIndices))
for i := range committeeIndices {
	indicesUint[i] = uint16(committeeIndices[i])
}
buf := new(bytes.Buffer)
if err = binary.Write(buf, binary.LittleEndian, indicesUint); err != nil {
	return Id{}, err
}
h = hash.Hash(append(dataHash[:], buf.Bytes()...))
```
Running `NewId` with 16 bits set
```
BenchmarkNewId-12         249559              5036 ns/op
```
Running the string version
```
BenchmarkNewId-12         275055              4297 ns/op
```
Do you have an idea how to make this faster?